src/operations/blur.jl

```diff
@@ -0,0 +1,38 @@
+import ImageFiltering: imfilter, KernelFactors.gaussian
+
+struct GaussianBlur{K <: AbstractVector, S <: AbstractVector} <: ImageOperation
+    k::K
+    σ::S
+
+    function GaussianBlur(k::K, σ::S) where {K <: AbstractVector{<:Integer},
+                                             S <: AbstractVector{<:Real}}
+        minimum(k) > 0 || throw(ArgumentError("Kernel size must be positive: $(k)"))
+        minimum(σ) > 0 || throw(ArgumentError("σ must be positive: $(σ)"))
+        new{K, S}(k, σ)
+    end
+end
+
+GaussianBlur(k, σ) = GaussianBlur(vectorize(k), vectorize(σ))
+
+randparam(op::GaussianBlur, img) = (safe_rand(op.k), safe_rand(op.σ))
+
+@inline supports_eager(::GaussianBlur) = true
+
+function applyeager(op::GaussianBlur, img::AbstractArray, (k, σ))
+    kernel = gaussian((σ, σ), (k, k))
```

Comment:
See also https://juliaimages.org/ImageFiltering.jl/dev/#Factored-kernels for a more efficient version of this; two 1-d gaussian blur is equivalent to one 2-d gaussian blur.

Reply:
I am using `KernelFactors.gaussian`, so eventually [this method](https://github.com/JuliaImages/ImageFiltering.jl/blob/f85e7b0ad4cd6fb8499ccdf25712e6f00be5b982/src/kernelfactors.jl#L427) gets invoked. I think it is what you are proposing, isn't it?

Reply:
Yeah, you're right, I missed that `using` statement.